src/sentry/search/django/backend.py

```diff
@@ -183,6 +183,32 @@ def assigned_to_filter(queryset, user, project):
     )
 
 
+def parse_release(project, environment, value):
+    from sentry.models import Release
+
+    if value == 'latest':
+        release_qs = Release.objects.filter(
+            organization_id=project.organization_id,
+            projects=project,
+        )
+
+        if environment is not None:
+            release_qs = release_qs.filter(
+                releaseprojectenvironment__environment__id=environment.id
+            )
+
+        value = release_qs.extra(select={
+            'sort': 'COALESCE(date_released, date_added)',
+        }).order_by('-sort').values_list(
+            'version', flat=True
+        ).first()
+
+        if value is None:
+            raise Release.DoesNotExist
```

Comment:
You can replace `.first()` with `[:1].get()` if you'd like to not have to explicitly raise here.